pkg/hpke/url.go

```diff
@@ -145,22 +145,31 @@ func withoutHPKEParams(values url.Values) url.Values {
 	return filtered
 }
 
-var zstdEncoder, _ = zstd.NewWriter(nil, zstd.WithEncoderLevel(zstd.SpeedBestCompression))
-
 func encodeQueryStringV1(values url.Values) []byte {
 	return []byte(values.Encode())
 }
 
+const zstdWindowSize = 8 << 10 // 8kiB
+
+var zstdEncoder, _ = zstd.NewWriter(nil,
+	zstd.WithEncoderLevel(zstd.SpeedBestCompression),
+	zstd.WithEncoderConcurrency(1),
```

Comment:
It looks like with this option only one call to `EncodeAll()` can happen at a time (it may block while acquiring the one available encoder instance, see [here](https://github.com/klauspost/compress/blob/8eb6542f08f086fb92796f6d8a4531d754947309/zstd/encoder.go#L495)).

Do we feel comfortable about that?